knock/src/pages/MainPage.tsx

```diff
@@ -1,11 +1,78 @@
+import MainPageInput from '../components/MainPageInput/MainPageInput';
+import Image from '../core/ui/CommonImage/Image';
 import MetaTags from '../core/ui/MetaTags/MetaTags';
+import MainPageLogo from '../core/assets/image/MainPgaeLogo.svg';
+import backgroundImg from '../core/assets/image/mainimage.png';
+import rightArrow from '../core/assets/icon/Hyphen-Arrow-right.svg';
+import styles from './MainPage.module.scss';
+import UButton from '../core/ui/buttons/UButton/UButton';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createSubject } from '../lib/api/Subject';
+import useUserInfo from '../lib/hooks/useUserInfo';
+
+const MainPage = () => {
+  const [name, setName] = useState('');
+  const navigate = useNavigate();
+  const { handleUserInfo } = useUserInfo();
+
+  const handleNameSet = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (name && e.key === 'Enter') {
+      handleSubmit(e);
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLElement>) => {
+    e.preventDefault();
+    try {
+      const { id, name: subName, imageSource } = await createSubject({ name });
+      navigate(`/post/${id}/answer`);
+      handleUserInfo({ id, name: subName, imageSource });
+    } catch (e) {
+      console.error(e);
+    }
+  };
 
-function MainPage() {
   return (
-    <>
+    <div className={`${styles['mainPage']}`}>
       <MetaTags />
-    </>
+      <UButton
+        type="box"
+        handleClick={handleSubmit}
+        isLightTheme={true}
+        className={`${styles['mainPage__answer']}`}
+      >
+        답변하러 가기
+        <Image
+          src={rightArrow}
+          alt="rightArrow"
+          imageClassName={styles['mainPage__answer__rightArrowImg']}
+        />
```

Comment:
아이콘 태그를 사용해보시는건 어떤가요??
```suggestion
        <Icon
          src={rightArrow}
          alt="rightArrow"
          imageClassName={styles['mainPage__answer__rightArrowImg']}
        />
```

Reply:
감사합니다!